File: 001. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  lookup = new Map();
  for (let i = 0; i < nums.length; i++) {
      if (lookup.has(nums[i])) {
          return [lookup.get(nums[i]), i];
      }
      lookup.set(target - nums[i], i);
  }
};

File: 001. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = dict()
        for i in range(len(nums)):
            diff = target - nums[i]
            if lookup.get(nums[i], -1) == -1:
                lookup[diff] = i
            else:
                return [lookup[nums[i]], i]